api/errors/server.go

```diff
@@ -0,0 +1,18 @@
+package errors
+
+import (
+	"net/http"
+
+	"github.com/docker/distribution/registry/api/errcode"
+)
+
+var (
+	// ErrorVersionMismatch is generated when we look for a container by
+	// name or ID and we can't find it.
+	ErrorVersionMismatch = errcode.Register(errGroup, errcode.ErrorDescriptor{
+		Value:          "VERSIONMISMATCH",
+		Message:        "%s",
```

Comment:
This shouldn't be "%s" because the main text is still hard coded in the source code, which defeats the purpose of creating this object. Create a new Error object for each unique string and only use "%s" for dynamic info (like the version number).